frontstage/controllers/notify_controller.py

```diff
@@ -0,0 +1,69 @@
+import logging
+from urllib import parse as urlparse
+
+import structlog
+
+from frontstage.exceptions import exceptions
+from requests_wrapper import Requests
+
+logger = structlog.wrap_logger(logging.getLogger(__name__))
+
+
+class NotifyGateway:
+    """ Client for Notify gateway"""
+
+    def __init__(self, config):
+        self.config = config
+        self.notify_url = config['RAS_NOTIFY_SERVICE_URL']
+        self.email_verification_template = config['RAS_NOTIFY_EMAIL_VERIFICATION_TEMPLATE']
+        self.request_password_change_template = config['RAS_NOTIFY_REQUEST_PASSWORD_CHANGE_TEMPLATE']
+        self.confirm_password_change_template = config['RAS_NOTIFY_CONFIRM_PASSWORD_CHANGE_TEMPLATE']
+        self.notify_account_locked = config['RAS_NOTIFY_ACCOUNT_LOCKED_TEMPLATE']
+
+    def _send_message(self, email, template_id, personalisation=None, reference=None):
+        """
+        Send message to gov.uk notify wrapper
+        :param email: email address of recipient
+        :param template_id: the template id on gov.uk notify to use
+        :param personalisation: placeholder values in the template
+        :param reference: reference to be generated if not using Notify's id
+        :rtype: 201 if success
+        """
```

Comment:
This should also have a `:raises KeyError:  Raised when the template name provided doesn't exist`

Reply:
Also add `:raises RasNotifyError: Raised when there is an error sending a message to the gov notify service`